lib/puppet/type/service.rb

```diff
@@ -60,6 +63,16 @@ module Puppet
         provider.manual_start
       end
 
+      # This only makes sense on systemd systems. Otherwise, it just defaults
+      # to disable.
+      newvalue(:mask, :event => :service_disabled) do
+        if provider.declared_feature?(:maskable)
```

Comment:
Puppet will automagically create predicate methods based on provider features declared in the type. So you can just do:

```
irb(main):002:0> require 'puppet'
=> true
irb(main):003:0> resource = Puppet::Type.type(:service).new(:name => 'foo')
=> ...
irb(main):004:0> resource.provider.maskable?
=> false
```